main.go

```diff
@@ -35,6 +35,7 @@ import (
 	"github.com/kubernetes-incubator/external-dns/controller"
 	"github.com/kubernetes-incubator/external-dns/pkg/apis/externaldns"
 	"github.com/kubernetes-incubator/external-dns/pkg/apis/externaldns/validation"
+	"github.com/kubernetes-incubator/external-dns/pkg/util/domains"
```

Comment:
why is this in the `util` ? it is generally not a good practice to put things into `util`, if it is part of the provider, it should be in the provider package